Add certificate_id field for for GCP CAS Enterprise tier

akos011221 opened the pull request:
<!--
1. If this is your first PR, please read our contributor guidelines
https://github.com/spiffe/spiffe/blob/master/CONTRIBUTING.md
https://github.com/spiffe/spire/blob/main/CONTRIBUTING.md

2. Please remember to include a DCO on every commit (`git commit -s`)
https://github.com/apps/dco
-->

**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
"gcp_cas" UpstreamAuthority plugin

**Description of change**
Add certificate_id field to CreateCertificateRequest to enable ENTERPRISE tier CA pool compatibility. The certificate_id is generated deterministically from the CSR public key hash for consistent resource naming.

**Which issue this PR fixes**
fixes #6180

sorindumitru commented on pkg/server/plugin/upstreamauthority/gcpcas/gcpcas.go:
```diff
@@ -284,9 +286,16 @@ func (p *Plugin) mintX509CA(ctx context.Context, csr []byte, preferredTTL int32)
+	// from a hash of the public key, ensuring the same certificate request always yields the same resource name.
+	certIDHash := sha256.Sum256(csrParsed.RawSubjectPublicKeyInfo)
+	// Truncate the 32-byte hash to 28 bytes (56 hex characters) to comply with the limit: 7 ("spiffe-") + 56 = 63.
+	certificateID := "spiffe-" + hex.EncodeToString(certIDHash[:28])
```
I'm not sure why we need this "spiffe-" prefix here. Keeping as much of the hash should make it easier for the id to be unique (although it's likely unique either way).

You could also use base32 instead of hex. I think either way it should be fine, but with base32 the sha256 hash will be encoded to less than 63 chars and will have the expected format. Either way, it might nicer to do the truncation at the end, maybe when adding the ID to the request object? `certificateID[:63]` makes it a bit easier to reason about than the `[:28]`, in my opinion.

akos011221 commented:
> Would it be possible to update the tests? Maybe the `fakeClient` can verify that a `certificateID` is passed in, has the right format and length, and hasn't been used before?

Added verification for the regex part for the format & length verification. To cover the "has it been used before" part of your request, I could came up with a solution to call the certificate minting multiple times, each with different CSR inside `TestGcpCAS()`.  Added a map to the fakeClient in which each Certificate ID is stored that the client seen so far, and for each certificate creation, the request's Certificate ID is verified against this map.

Also, switched all `require` inside `CreateCertificate` to `assert`, because I've noticed that if the `require` fails, the test will hang. https://pkg.go.dev/github.com/stretchr/testify/require?#hdr-Assertions

Please let me know if I should change anything.
